lldb/include/lldb/API/SBAddressRangeList.h

```diff
@@ -0,0 +1,58 @@
+//===-- SBAddressRangeList.h ------------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLDB_API_SBADDRESSRANGELIST_H
+#define LLDB_API_SBADDRESSRANGELIST_H
+
+#include <memory>
+
+#include "lldb/API/SBDefines.h"
+
+class AddressRangeListImpl;
+
+namespace lldb {
+
+class LLDB_API SBAddressRangeList {
+public:
+  SBAddressRangeList();
+
+  SBAddressRangeList(const lldb::SBAddressRangeList &rhs);
+
+  ~SBAddressRangeList();
+
+  const lldb::SBAddressRangeList &
+  operator=(const lldb::SBAddressRangeList &rhs);
+
+  uint32_t GetSize() const;
+
+  void Clear();
+
+  bool GetAddressRangeAtIndex(uint64_t idx, SBAddressRange &addr_range);
```

Comment:
Why doesn't this just return `SBAddressRange`? There are then 2 ways to check the validity of the range (the return value and what's in `addr_range`).

Also, why is the index `uint64_t` but the return value of GetSize is `uint32_t`? Seems like both are related to the possible indices of the list, we should pick one and use it everywhere.

Reply:
yes, please make this:
```
SBAddressRange GetAddressRangeAtIndex(uint64_t idx);
```
Easier to use in code and in python. Return a default constructed `SBAddressRange` if `idx` is out of range.